pkgs/test_api/lib/src/backend/test_location.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// The location of a test or group.
+class TestLocation {
```

Comment:
I originally just added a line/col to the `GroupEntry` class, but it turns out we need uri too, so it felt better to wrap this up in a class (so you either provide the whole lot or none at all).

We need a URI because filtering by the stack trace, we can say "if any frame of the call stack matches the requested file+line+col", but when using the location override, we wouldn't have a uri/path to match on.

This does mean we need to ensure this class is exported and available to any callers of `test()`.

Reply:
I do think wrapping this all in a class makes sense, fewer new arguments to group/test as well.